mmv1/third_party/terraform/services/container/node_config.go.tmpl

```diff
@@ -899,6 +901,46 @@ func schemaNodeConfig() *schema.Schema {
 	}
 }
 
+
+func schemaBootDiskConfig() *schema.Schema {
+	return &schema.Schema{
+		Type: schema.TypeList,
+		Optional: true,
+		Computed: true,
+		MaxItems: 1,
+		Description: `Boot disk configuration for node pools nodes.`,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"disk_type": {
```

Comment:
Should this be required? If someone specifies a boot_disk it seems like they should be required to set the type and size

Reply:
The primary reason is that changing this would be backwards breaking. (People using the legacy field would be forced to update their configs.) That said, there are defaults that the API puts into place, making this non-required.

Reply:
I'm not sure how this would be breaking

Adding a required field as a sub-field of a new field isn't breaking because it only applies if the top-level field is specified. So disk_type would only be required if the user specifies boot_disk, which they can't currently.